File: drawFunc.py
import numpy as np
import matplotlib.pyplot as plt
from func import *

def draw(f_x):
# allocate grids
    dx = np.arange(-3.0, 2.5, .02)
    dy = np.arange(-4.5, 4.5, .02)
    X, Y = np.meshgrid(dx, dy)

# eval function f_x
    Z = [f_x(np.matrix([[x1], [x2]])) \
        for (x1, x2) in zip(np.hstack(X), np.hstack(Y))]
    Z = np.array(Z)
    Z = Z.reshape(np.shape(X))

# take this pencil
    levels = [0.0, 1, 2, 5, 20, 30, 50, 80, 100, 200, 400]
    #cs = plt.contour(X, Y, Z, levels, colors='k')
    cs = plt.contourf(X, Y, Z, 20, cmap=plt.cm.YlGnBu)
    #plt.clabel(cs, fmt='%.1f', inline=1)

File: jl/DrawFunc.jl
module DrawFunc

using TargetFunction
using PyCall
@pyimport matplotlib.pyplot as plt
@pyimport matplotlib.cm as cm

function draw2D(fun::TargetFunction.ObjFun)

    dx = [-3.0:.02:2.5]
    dy = [-4.5:.02:4.5]'

    m, n = length(dx), length(dy)
    dx = repmat(dx, 1, n)[:]
    dy = repmat(dy, m, 1)[:]
    z = [fun.f_x([i;j]) for (i, j) in zip(dx, dy)]

    dx, dy, z = reshape(dx, m, n), reshape(dy, m, n), reshape(z, m, n)
    cs = plt.contourf(dx, dy, z, 20, cmap=cm.get_cmap("YlGnBu"))
end

end
